src/tests/integration/test_connection.py

```diff
@@ -199,3 +210,184 @@ def test_dbapi_connection_sets_right_dialect(dynamic_db, dialect, request):
+    monkeypatch.setattr(conn._connection, "commit", conn_mock_commit)
+
+    conn.raw_execute(
+        """
+CREATE TABLE numbers (
+    x INTEGER
+);
+"""
+    )
+    conn.raw_execute(
+        """
+    INSERT INTO numbers VALUES (1), (2), (3);
+    """
+    )
+
+    # if commit is working, the table should be readable from another connection
+    another = SQLAlchemyConnection(
+        engine=create_engine("duckdb:///my.db"), config=Config
+    )
+    another_mock_commit = Mock(wraps=another._connection.commit)
+    monkeypatch.setattr(another._connection, "commit", another_mock_commit)
+
+    results = another.raw_execute("SELECT * FROM numbers")
+
+    assert list(results) == [(1,), (2,), (3,)]
+    conn_mock_commit.assert_has_calls([call(), call()])
+    # due to https://github.com/Mause/duckdb_engine/issues/734, we should not
+    # call commit on SELECT statements
+    another_mock_commit.assert_not_called()
+
+
+def test_postgres_autocommit_on_with_manual_commit(setup_postgreSQL, monkeypatch):
+    url = _testing.DatabaseConfigHelper.get_database_url("postgreSQL")
+
+    class Config:
+        autocommit = True
+
+    monkeypatch.setattr(
+        connection, "set_sqlalchemy_isolation_level", Mock(return_value=False)
+    )
+
+    engine = create_engine(url)
+
+    conn = SQLAlchemyConnection(engine=engine, config=Config)
+    conn_mock_commit = Mock(wraps=conn._connection.commit)
+    monkeypatch.setattr(conn._connection, "commit", conn_mock_commit)
+
+    conn.raw_execute(
+        """
+CREATE TABLE numbers (
+    x INTEGER
+);
+"""
+    )
+    conn.raw_execute(
+        """
+    INSERT INTO numbers VALUES (1), (2), (3);
+    """
+    )
+
+    # if commit is working, the table should be readable from another connection
+    another = SQLAlchemyConnection(engine=create_engine(url), config=Config)
+    another_mock_commit = Mock(wraps=another._connection.commit)
+    monkeypatch.setattr(another._connection, "commit", another_mock_commit)
+
+    results = another.raw_execute("SELECT * FROM numbers")
+
+    assert list(results) == [(1,), (2,), (3,)]
+    conn_mock_commit.assert_has_calls([call(), call()])
+    # due to https://github.com/Mause/duckdb_engine/issues/734, we are not calling
+    # commit on SELECT statements for DuckDB, but for other databases we do
+    another_mock_commit.assert_has_calls([call()])
+
+
+def test_duckdb_autocommit_off(tmp_empty):
+    class Config:
+        autocommit = False
+
+    engine = create_engine("duckdb:///my.db")
+    conn = SQLAlchemyConnection(engine=engine, config=Config)
+    conn.raw_execute(
+        """
+CREATE TABLE numbers (
+    x INTEGER
+);
+"""
+    )
+    conn.raw_execute(
+        """
+    INSERT INTO numbers VALUES (1), (2), (3);
+    """
+    )
+
+    # since autocommit is off, the table should not be readable from another connection
+    another = SQLAlchemyConnection(
+        engine=create_engine("duckdb:///my.db"), config=Config
```

Comment:
Do we need to pass the config here again? it shouldn't be readable anyways right?

Reply:
we have to pass config now because I moved the autocommit feature from run.py to connection.py. I don't like this design much so I'll do a refactor of the whole config reading, but I'm passing it for now

> it shouldn't be readable anyways right?

if autocommit is off, it shouldn't be readable. if it's on, it should be readable